Merge new-security-fragment into master

MachoBandido08 opened the pull request:
Étant donné que la fonctionnalité de télécharger la description des types d'urgence n'est pas si importante, je ne l'ai pas mis la fonctionnalité de télécharger la description du type d'urgence. Je le ferai plus tard.
![20190205_133034](https://user-images.githubusercontent.com/44484236/52320369-fabf7880-299c-11e9-8eff-daa24b746c42.gif)

JasonNarrainen committed:
Added security fragment and navigation to it from ets fragment

MachoBandido08 committed:
Added security adapter, security model, security item layout and an example layout

MachoBandido08 committed:
Variable name change

MachoBandido08 committed:
Setup RecyclerView

JasonNarrainen committed:
Put the recycler view in comments

MachoBandido08 committed:
SecurityModel deleted and SecurityDetail added

MachoBandido08 committed:
Modification added

MachoBandido08 committed:
DetailSecurity implemented with webview

MachoBandido08 committed:
Html files created

MachoBandido08 committed:
Little fix

MachoBandido08 committed:
Fix layout margin bottom

MachoBandido08 committed:
Call button implemented and layout fix

MachoBandido08 committed:
Security header layout added

MachoBandido08 committed:
Nested scroll fixed

MachoBandido08 committed:
Toolbar set to not collapse

MachoBandido08 committed:
Google Map implemented

* Location and markers not set to ETS
* NestedScrollView obstructs Map's scroll and zoom

MachoBandido08 committed:
Disable nestedScrollView's touch events when map's touch events are active

MachoBandido08 committed:
Set map camera and marker location to ETS

MachoBandido08 committed:
Format fragment and layout code

MachoBandido08 committed:
Format and remove unnecessary code

MachoBandido08 committed:
Merge branch 'master' of https://github.com/ApplETS/Notre-Dame into merge_master_into_security_safeargs

MachoBandido08 committed:
Merge master into new-security-fragment

- Use androidx.navigation.safeargs.kotlin plugin because the Android app should only use Kotlin

MachoBandido08 committed:
Fix mapView memory leak

* NestedScrollView's memory leak may be fixed
* All security module formatted

MachoBandido08 committed:
More format code

MachoBandido08 committed:
Add transition in Security

MachoBandido08 committed:
Modify gradle for google api keys

MachoBandido08 committed:
Encrypt google maps api keys

MachoBandido08 committed:
Revert commit "Encrypt google maps api keys"

This reverts commit fc1147452d2815524289dbbde0b8fe3c58a0b9f2.

MachoBandido08 committed:
Create xml values files for different build/flavor

* Remove gradle code

MachoBandido08 committed:
Encrypt api keys

Added new api keys for production

MachoBandido08 committed:
Test ktlintFormat and ktlintCheck

MachoBandido08 committed:
Merge branch 'master' into new-security-fragment

MachoBandido08 committed:
Format with ktlint

MachoBandido08 committed:
Format again with ktlint

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+
+            <ImageView
+                android:id="@+id/imageView3"
+                android:layout_width="45dp"
```
Dans les spécifications de Material Design, les [listes à 2 éléments](https://material.io/design/components/lists.html?image=https%3A%2F%2Fstorage.googleapis.com%2Fspec-host-backup%2Fmio-design%252Fassets%252F1-YqqvjVvKr2aWHg6yQ9KOP9yFbk-4o9t%252Flist-spec-doubleline-individual.png#specs) doivent avoir des icônes de 40 dp en hauteur et en largeur.

zaclimon commented on android/app/src/main/res/layout/fragment_security_detail.xml:
```diff
@@ -0,0 +1,23 @@
+        android:layout_alignParentBottom="true"
+        android:layout_centerHorizontal="true"
+        android:text="@string/emergency_call"
+        android:textAppearance="?android:attr/textAppearanceLarge" />
```
Je ne crois pas que le texte a nécessairement besoin d'être gros. Le système peut normalement ajuster le texte en fonction des paramètres de l'utilisateur. Tu peux utiliser le composant [MaterialButton](https://material.io/develop/android/components/material-button/) pour avoir les bonnes dimensions.

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+                android:id="@+id/txtJoindreSecurite"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:padding="10dp"
```
Les subheaders sont supposés avoir 16 dp de marge/padding. De cette manière, ça sera aligné avec les listes. Tu peux voir les specs que j'avais laissé sur mon dernier commentaire.

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="16dp"
+                android:padding="10dp"
```
Same as above.

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+                app:layout_constraintTop_toBottomOf="@+id/txtJoindreSecurite" />
+
+            <TextView
+                android:id="@+id/textView3"
```
Je crois qu'il serait mieux de changer le id parce qu'il est un peu difficile de déterminer quelle est la section représenté par le `TextView`.

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+                app:layout_constraintTop_toTopOf="@+id/imageView" />
+
+            <TextView
+                android:id="@+id/textView5"
```
Same as above

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+                app:layout_constraintTop_toTopOf="@+id/imageView3" />
+
+            <TextView
+                android:id="@+id/textView6"
```
Same as above.

zaclimon commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,141 @@
+                app:layout_constraintTop_toBottomOf="@+id/textView5" />
+
+            <TextView
+                android:id="@+id/textView4"
```
Same as above.

zaclimon reviewed (changes requested):
À l'exception des changements que j'ai remarqués (Incluant les marges pour le `RecyclerView` aussi) nice job! L'intégration de la carte est :fire:.

@Hamtara Is that how you were seeing it?

Sonphil reviewed (changes requested):
J'ai remarqué que la barre demeurait cachée lorsqu'on se rendait à ce fragment. Tu dois utiliser la ligne suivante `appBarLayout.setExpanded(true, true)` pour afficher la barre correctement.
![image](https://user-images.githubusercontent.com/22182973/52418793-d5c22700-2abc-11e9-87d1-98fc750f3f79.png)

Aussi, la barre devrait avoir le même comportement au défilement que les autres pages. Je crois que tu dois simplement ajouter le comportement à ta vue défilante avec la ligne suivante : `app:layout_behavior="@string/appbar_scrolling_view_behavior"`

Sonphil commented on android/app/src/main/kotlin/ca/etsmtl/applets/etsmobile/presentation/security/CustomMapView.kt:
```diff
@@ -0,0 +1,27 @@
+        context,
+        attrs,
+        defStyle
+    )
```
En Kotlin, tu n'as besoin de définir plusieurs constructeurs pour les custom views notamment grâce aux [valeurs par défaut](https://try.kotlinlang.org/#/Kotlin%20Koans/Introduction/Default%20arguments/Task.kt).

Tout d'abord, enlève tes constructuers et l'IDE te signelera une erreur. Par la suite, utilise la combinaison de touch ALT + Entrée pour afficher le menu suivant.
![image](https://user-images.githubusercontent.com/22182973/52420066-508c4180-2abf-11e9-883c-65d7a01b3664.png)
Sélectionne l'option qui  te permet d'obtenir le résultat suivant : 
![image](https://user-images.githubusercontent.com/22182973/52419840-e83d6000-2abe-11e9-84f9-0d726ae43649.png)

Sonphil commented on android/app/src/main/kotlin/ca/etsmtl/applets/etsmobile/presentation/security/SecurityDetailFragment.kt:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    private fun setToolbarTitle() {
+        (activity as? MainActivity)?.toolbar!!.title = args.securityName
```
Le title ne pourrait pas être changé en utilisant le xml du graph?

J'ai fait une recherche rapide et j'ai vu la solution suivante : https://stackoverflow.com/questions/50599238/dynamic-actionbar-title-from-a-fragment-using-androidx-navigation

MachoBandido08 replied:
merci

MachoBandido08 committed:
Integrate Material Design and other minor changes in xml files

MachoBandido08 committed:
Change setTitle with graph and another minor change

Sonphil commented on android/app/src/main/res/layout/fragment_security.xml:
```diff
@@ -0,0 +1,142 @@
+            android:nestedScrollingEnabled="false"
+            android:scrollbars="none" />
+
+    </LinearLayout>
```
Ce `linearLayout` est-il nécessaire. Tu ne pourrais pas déplacer la `MapView` et le `RecyclerView` dans le `ConstraintLayout` et éviter un niveau d'imbrication?

zaclimon committed:
Merge branch 'master' into new-security-fragment

Sonphil commented on android/app/build.gradle:
```diff
@@ -103,6 +104,7 @@ dependencies {
 
     // Architecture components
     implementation "androidx.lifecycle:lifecycle-extensions:$lifecycle"
+    implementation 'com.google.android.gms:play-services-maps:16.0.0'
```
Ce n'est pas un architecture component. Pourrais-tu déplacer cette ligne?

zaclimon replied:
Aussi, ajouter une variable pour la version de la librairie serait nice.

MachoBandido08 committed:
Create a new toolbar for security detail and other fixes.

I will modify it later. It also fixes the button "Emergency call" in SecurityDetail who couldn't fit entirely in the bottom of the screen.

- Add appBarLayout.setExpanded
- Fix the toolbar who couldn't be scrollable in security and security detail fragments
- Disable transitions between security and security detail fragment because they interfere badly with the appBarLayout's animations. I will add a custom transition for them later.
- Other minor changes

MachoBandido08 committed:
Move map view and recyclerview inside constraintLayout

MachoBandido08 committed:
Move google maps dependency and add variable to it

MachoBandido08 committed:
Merge remote-tracking branch 'origin/new-security-fragment' into new-security-fragment

MachoBandido08 commented:
@Sonphil Ai-je bien accomplis tes demandes?
